Java/ds/Binary_Tree.java

```diff
@@ -37,14 +37,29 @@ public Node takeInput(Node parent,Boolean isleftorright){
         if(choice){
             node.right=takeInput(node,false);
         }
+        System.out.println("Do you want tree traversal ");
+        boolean chose = s.nextBoolean();
+        if(chose){
+            traversal();
+        }
         return node;
     }
+    public void traversal()
+    {
+        System.out.println("Enter 1 for preorder traversal or enter 2 for postorder traversal ");
+        int option = s.nextInt();
+        switch(option)
+        {
+            case 1:
+            preOrder(root);
+            break;
 
-    public void preOrder() {
+            case 2:
+            postOrder(root);
+            break;
+        }
 
-        preOrder(root);
     }
-
     private void preOrder(Node node) {
         if (node == null) {
```

Comment:
The preorder traversal is being assigned to another participant in JAVA. I would like you to add a new file containing your entire postorder code in it instead of adding the code in the existing file.

Reply:
@Abhijit2505 i will do the changes requested

Reply:
@Abhijit2505 @routrohan please review this pull request with changes done